packages/find-workspaces-root/src/index.ts

```diff
@@ -30,13 +27,21 @@ export default async function findWorkspacesRoot(cwd: string): Promise<string> {
         }
       }
     },
-    { cwd, type: "directory" }
+    { cwd: options.cwd, type: "directory" }
   );
-  if (firstPkgJsonDirectory === undefined) {
-    throw new NoPkgJsonFound(cwd);
+
+export default async function findWorkspacesRoot(cwd: string): Promise<string> {
+  let workspacesRoot = await findWorkspacesUp({ cwd });
+
+  if (workspacesRoot) {
+    return workspacesRoot;
   }
-  if (dir === undefined) {
-    return firstPkgJsonDirectory;
+
+  let singlePackageRoot = await findWorkspacesUp({ cwd, tools: ["root"] });
```

Comment:
I'm quite unsure why this is needed (but there is an explicit test for it). `manypkg` is by definition supposed to help with validating monorepo setups, so i dont understand why a single package mode is supported?

Reply:
It exists so that other tools can use this even though it's not Manypkg's use case. (I have some ideas in mind where this will be necessary)

Reply:
> I have some ideas in mind where this will be necessary

Could you share them? I'm trying to understand the whole context, so that would be helpful to me

Reply:
The main one I have right now is a minimal wrapper around npm, pnpm, Yarn and possibly Bolt that chooses the right tool given a repo and I want it to work on single package repos as well.